R/namespace.R

```diff
@@ -238,6 +238,14 @@ roxy_tag_parse.roxy_tag_importFrom <- function(x) {
 }
 #' @export
 roxy_tag_ns.roxy_tag_importFrom <- function(x, block, env, import_only = FALSE) {
+  pkg <- x$val[1L]
+  if (requireNamespace(pkg, quietly = TRUE)) {
+    importing <- x$val[-1L]
+    unknown_idx <- !importing %in% getNamespaceExports(pkg)
+    if (any(unknown_idx)) {
+      warn_roxy_tag(x, "Unknown {cli::qty(sum(unknown_idx))} export{?s} in {.code @importFrom {pkg}}: {.code {importing[unknown_idx]}}")
```

Comment:
You'll also need to modify `x$val` here to avoid generating the bad input.

Reply:
i.e. just drop the unknown exports after throwing the warning?

great idea! makes me much happier with a warning instead of an error

Reply:
Yeah, exactly. I'd say that's the general roxgen2 strategy for problematic user input.